Source/Tests/Scene/Node.cpp

```diff
@@ -44,3 +45,136 @@ TEST_CASE("Load node from XML node file")
+    auto childComponent = child->CreateComponent<StaticModel>();
+    auto grandChildComponent = grandChild->CreateComponent<AnimatedModel>();
+
+    ea::vector<Component*> dest;
+    node->FindComponents(dest, StaticModel::TypeId, ComponentSearchFlag::Self);
+    CHECK(dest == ea::vector<Component*>{nodeComponent});
+
+    node->FindComponents(dest, Drawable::TypeId, ComponentSearchFlag::Self | ComponentSearchFlag::Derived);
+    CHECK(dest == ea::vector<Component*>{nodeComponent});
+
+    node->FindComponents(dest, StaticModel::TypeId, ComponentSearchFlag::Parent);
+    CHECK(dest == ea::vector<Component*>{parentComponent});
+
+    node->FindComponents(dest, StaticModel::TypeId, ComponentSearchFlag::ParentRecursive);
+    CHECK(dest == ea::vector<Component*>{parentComponent});
+
+    node->FindComponents(dest, StaticModel::TypeId, ComponentSearchFlag::ParentRecursive | ComponentSearchFlag::Derived);
+    CHECK(dest == ea::vector<Component*>{parentComponent, rootComponent});
+
+    node->FindComponents(dest, StaticModel::TypeId, ComponentSearchFlag::Self | ComponentSearchFlag::ParentRecursive);
+    CHECK(dest == ea::vector<Component*>{nodeComponent, parentComponent});
+
+    node->FindComponents(dest, Drawable::TypeId, ComponentSearchFlag::Self | ComponentSearchFlag::ParentRecursive | ComponentSearchFlag::Derived);
+    CHECK(dest == ea::vector<Component*>{nodeComponent, parentComponent, rootComponent});
+
+    node->FindComponents(dest, StaticModel::TypeId, ComponentSearchFlag::Children);
+    CHECK(dest == ea::vector<Component*>{childComponent});
+
+    node->FindComponents(dest, StaticModel::TypeId, ComponentSearchFlag::ChildrenRecursive);
+    CHECK(dest == ea::vector<Component*>{childComponent});
+
+    node->FindComponents(dest, StaticModel::TypeId, ComponentSearchFlag::ChildrenRecursive | ComponentSearchFlag::Derived);
+    CHECK(dest == ea::vector<Component*>{childComponent, grandChildComponent});
+
+    node->FindComponents(dest, StaticModel::TypeId,
+        ComponentSearchFlag::Self | ComponentSearchFlag::ChildrenRecursive | ComponentSearchFlag::Derived);
+    CHECK(dest == ea::vector<Component*>{nodeComponent, childComponent, grandChildComponent});
+};
+
+
+TEST_CASE("Test GetComponents<T>")
+{
+    auto context = Tests::GetOrCreateContext(Tests::CreateCompleteContext);
+    auto scene = MakeShared<Scene>(context);
+
+    auto root = scene->CreateChild("Root");
+    auto parent = root->CreateChild("Parent");
+    auto node = parent->CreateChild("Node");
+    auto child = node->CreateChild("Child");
+    auto grandChild = child->CreateChild("GrandChild");
+
+    auto rootComponent = root->CreateComponent<AnimatedModel>();
+    auto parentComponent = parent->CreateComponent<StaticModel>();
+    auto nodeComponent = node->CreateComponent<StaticModel>();
+    auto childComponent = child->CreateComponent<StaticModel>();
+    auto grandChildComponent = grandChild->CreateComponent<AnimatedModel>();
+
+    ea::vector<Component*> dest;
```

Comment:
Can we test/support vector of WeakPtrs too? It looks like reasonable use case and it should not be too hard.